File: frontend/src/core/models/video.ts
interface VideoResponseData {
  id: number;
  title: string;
  slug: string;
}

class Video {
  id: number;
  title: string;
  slug: string;

  constructor(data: VideoResponseData) {
    this.id = data.id;
    this.title = data.title;
    this.slug = data.slug;
  }
}

export default Video;

File: backend/videos/models/video.py
from django.db import models

from videos.models.collection import Collection


class Video(models.Model):
    title = models.CharField(max_length=200)
    slug = models.SlugField(max_length=200, unique=True)
    collections = models.ManyToManyField(Collection, related_name='videos')

    def __str__(self):
        return self.title
